### PR TITLE
app and secret changes to AS3

### PR DESCRIPTION
**Pull request by ruth0048**

### CODE REVIEW

File: UnitTests/src/Main.as [hunk 1]
```diff
@@ -16,13 +18,15 @@ package
...
-        public var APP_ID:String = "";
-        public var SECRET:String = "";
-        public var APP_VERSION:String = "";
-        public var SERVER_URL:String = "";
-        public var PARENT_LEVEL_NAME:String = "";
-        public var CHILD_APP_ID:String = "";
-        public var PEER_NAME:String = "";
+        public var APP_ID:String = "20001";
+        public var SECRET:String = "4e51b45c-030e-4f21-8457-dc53c9a0ed5f";
+        public var CHILD_SECRET:String = "f8cec1cf-2f95-4989-910c-8caf598f83db";
```
Comment (ruth0048) [inline UnitTests/src/Main.as]: I added this manually, as naturally this is read of an ids, but our ids isn't up to date to include this secret.

File: UnitTests/src/Main.as [hunk 2]
```diff
@@ -41,17 +45,29 @@ package
             trace("--- Unit Tests ---");
 
             loadIds();
+
+            //temporary fix because of ids parsing issues. 
+            if(CHILD_SECRET == "undefined")
+            {
+                //hard coded from test childapp on internal
+                CHILD_SECRET = "f8cec1cf-2f95-4989-910c-8caf598f83db";
```
Comment (davidstl) [inline UnitTests/src/Main.as]: ? Don't hardcode this its visible to the public.

File: brainCloudClient/src/com/bitheads/braincloud/BrainCloudClient.as [hunk 3]
```diff
@@ -134,12 +135,46 @@ package com.bitheads.braincloud
...
+         */
+        public function initializeWithApps(appId:String, secretMap:Dictionary, appVersion:String, serverUrl:String = DEFAULT_SERVER_URL):void
+        {
+            var error:String = null;            
+            if (isNullOrEmpty(serverUrl))
+                error = "serverUrl was null or empty";
+            else if (isNullOrEmpty(appId))
+                error = "appId was null or empty";
+            else if (isNullOrEmpty(secretMap[appId]))
+                error = "secretKey was null or empty";
```
Comment (davidstl) [inline brainCloudClient/src/com/bitheads/braincloud/BrainCloudClient.as]: "no matching secret for appId"

File: brainCloudClient/src/com/bitheads/braincloud/comms/BrainCloudComms.as [hunk 4]
```diff
@@ -134,6 +135,19 @@ package com.bitheads.braincloud.comms
             _url = serverUrl;
         }
         
+        public function initializeWithApps(appId:String, secretMap:Dictionary, serverUrl:String):void
```
Comment (davidstl) [inline brainCloudClient/src/com/bitheads/braincloud/comms/BrainCloudComms.as]: Normal initialize call should create the map with 1 value.

File: brainCloudClient/testsrc/Main.as [hunk 5]
```diff
@@ -21,15 +22,22 @@ public class Main extends Sprite
...
  
 	var bc:BrainCloudClient;
 	
     public function Main()
     {
 		trace("Main");
 		
+		SECRET_MAP = new Dictionary;
+		SECRET_MAP[APP_ID] = SECRET;
+
```
Comment (davidstl) [inline brainCloudClient/testsrc/Main.as]: Your error is probably here. You didnt set the child app id + secret.

Comment (davidstl) [reply brainCloudClient/testsrc/Main.as]: wait what is this file. Which one of the 2 mains is used? probably one of those is not used be careful

File: UnitTests/src/Main.as [hunk 6]
```diff
@@ -43,15 +47,20 @@ package
             loadIds();
             
             // ChatTest.createTests(m_tests); // we don't officially support RTT features in AS3
+            IdentityTest.createTests(m_tests);
+            //GroupTest.createTests(m_tests);
```
Comment (davidstl) [inline UnitTests/src/Main.as]: uncomment before merging

File: brainCloudClient/src/com/bitheads/braincloud/comms/BrainCloudComms.as [hunk 7]
```diff
@@ -414,6 +432,17 @@ package com.bitheads.braincloud.comms
                     {
                         _profileId = data.profileId;
                     }
+                    if (data.switchToAppId != null)
+                    {
+                        _fnDebugOutput("Current secret: " + _secret);
```
Comment (davidstl) [inline brainCloudClient/src/com/bitheads/braincloud/comms/BrainCloudComms.as]: Dont log secret. Remove those debug log.

File: brainCloudClient/testsrc/Main.as [hunk 8]
```diff
@@ -27,8 +27,10 @@ public class Main extends Sprite
     public function Main()
     {
 		trace("Main");
-		
+
 		bc = new BrainCloudClient();
+		
+		//bc.initialize(APP_ID, SECRET, APP_VERSION);
```
Comment (davidstl) [inline brainCloudClient/testsrc/Main.as]: ?

**Pull request merged**